### PR TITLE
Make LFVM OpCodes binary compatible with Tosca definitions. 

### PR DESCRIPTION
**Pull request by LuisPH3**
This PR makes OpCodes in `lfvm` package have the same value as OpCodes defined in `Tosca/vm` package. 

The PR is rather large because any test or facility that relied on consecutive Opcodes had to be rewritten to allow for the sparse nature of ETH opcodes. 

Any opcode with value less or equal to 0xFF will be reserved for the opcodes declared in the `tosca/vm` package, and they may be equivalent to the Ethereum definition. 
OpCodes starting at value 0x100 are reserved for the LFVM extended set of instructions. Which includes both required opcodes to function and super-instructions. 

### The benefit: 
- Simplifies translation and makes experience more intuitive, opcodes can be cast from Tosca into LFVM. 

### Cons:
- The existing opcodes were fine-tuned for better dispatcher switch performance, this is discussed in the benchmarks section ahead.

### Performance  
```
goos: linux
goarch: amd64
pkg: github.com/Fantom-foundation/Tosca/go/integration_test/interpreter
cpu: Intel(R) Xeon(R) Platinum 8481C CPU @ 2.70GHz
                         │   main_20   │         unified_opcodes_20         │
                         │   sec/op    │   sec/op     vs base               │
Inc/1/lfvm-8               1.672µ ± 0%   1.674µ ± 0%       ~ (p=0.149 n=20)
Inc/10/lfvm-8              1.670µ ± 0%   1.672µ ± 0%       ~ (p=0.941 n=20)
Fib/1/lfvm-8               1.434µ ± 0%   1.461µ ± 0%  +1.85% (p=0.000 n=20)
Fib/5/lfvm-8               9.033µ ± 0%   9.023µ ± 0%  -0.12% (p=0.007 n=20)
Fib/10/lfvm-8              94.89µ ± 0%   94.00µ ± 0%  -0.94% (p=0.000 n=20)
Fib/15/lfvm-8              1.047m ± 0%   1.035m ± 0%  -1.09% (p=0.000 n=20)
Fib/20/lfvm-8              11.58m ± 0%   11.45m ± 0%  -1.13% (p=0.000 n=20)
Sha3/1/lfvm-8              636.1n ± 0%   633.6n ± 0%  -0.41% (p=0.000 n=20)
Sha3/10/lfvm-8             1.937µ ± 0%   1.957µ ± 0%  +1.03% (p=0.000 n=20)
Sha3/100/lfvm-8            14.55µ ± 0%   14.50µ ± 0%  -0.38% (p=0.000 n=20)
Sha3/1000/lfvm-8           137.5µ ± 0%   137.6µ ± 0%  +0.09% (p=0.011 n=20)
Arith/1/lfvm-8             1.823µ ± 0%   1.824µ ± 0%       ~ (p=0.773 n=20)
Arith/10/lfvm-8            4.812µ ± 0%   4.834µ ± 0%  +0.47% (p=0.000 n=20)
Arith/100/lfvm-8           34.17µ ± 0%   34.20µ ± 0%  +0.09% (p=0.038 n=20)
Arith/280/lfvm-8           95.20µ ± 0%   95.29µ ± 0%       ~ (p=0.071 n=20)
Memory/1/lfvm-8            2.942µ ± 0%   2.948µ ± 0%  +0.22% (p=0.043 n=20)
Memory/10/lfvm-8           7.121µ ± 0%   7.163µ ± 0%  +0.59% (p=0.000 n=20)
Memory/100/lfvm-8          48.58µ ± 0%   48.48µ ± 0%  -0.21% (p=0.002 n=20)
Memory/1000/lfvm-8         456.7µ ± 0%   455.4µ ± 0%  -0.30% (p=0.000 n=20)
Memory/10000/lfvm-8        4.538m ± 0%   4.511m ± 0%  -0.58% (p=0.000 n=20)
Analysis/jumpdest/lfvm-8   426.9n ± 0%   425.8n ± 0%       ~ (p=0.109 n=20)
Analysis/stop/lfvm-8       426.4n ± 0%   427.6n ± 0%       ~ (p=0.171 n=20)
Analysis/push1/lfvm-8      427.7n ± 0%   428.3n ± 0%       ~ (p=0.713 n=20)
Analysis/push32/lfvm-8     427.0n ± 0%   427.3n ± 0%       ~ (p=0.457 n=20)
geomean                    12.48µ        12.48µ       -0.01%

```

Meanwhile, does not suffer slowdown. 
image

part of #750 

TODO:
- [x] use 9 bit opcode definition everywhere
- [x] factor out opcode extra constants 
- [x] evaluate performance again,
- [x] CT

### CODE REVIEW

File: go/interpreter/lfvm/instruction_test.go [hunk 1]
```diff
@@ -13,15 +13,27 @@ package lfvm
...
 func TestInstruction_String(t *testing.T) {
-	instruction := Instruction{opcode: STOP, arg: 0x0000}
-	if got, want := instruction.String(), "STOP"; got != want {
-		t.Errorf("Instruction.String() = %q, want %q", got, want)
+
+	tests := []struct {
+		instruction Instruction
+		want        string
+	}{
+		{Instruction{opcode: STOP, arg: 0x0000}, "STOP"},
```
Comment (facuMH) [inline go/interpreter/lfvm/instruction_test.go:21]: I thought we were going to start using stringer generated code to prevent this kind of testing, no?

**Review by facuMH — APPROVED** [review]
awesome work !!! thanks for the refactor

File: go/interpreter/lfvm/instruction_test.go (hunk 1 above)
Comment (LuisPH3) [reply go/interpreter/lfvm/instruction_test.go:21]: I guess, but there is no infrastructure so far, this is just migrating the opcodes.

File: go/interpreter/lfvm/opcode.go [hunk 2]
```diff
@@ -497,3 +352,25 @@ func (o OpCode) decompose() []OpCode {
...
+// An executable OpCode is a valid OpCode that executes an operation; all valid
+// OpCodes with the exception of DATA, and NOOP are executable.
+// - JUMPDEST is an executable OpCode.
+// - INVALID is not an executable OpCode, since it is neither Valid
+func (op OpCode) isExecutable() bool {
+	if op < FIRST_LFVM_EXTENDED_OPCODE {
+		return op.isValid()
+	}
+	return op < END_LFVM_EXECUTABLE_OPCODES
+}
```
Comment (HerbertJordan) [inline go/interpreter/lfvm/opcode.go:376]: I think these functions can be eliminated by defining the following:
- OpCodes are 9-bit values; the first 7 bits are to be ignored, only the trailing 9 bits are relevant
- every possible OpCode is a "valid" opcode (the full range of 0-512)
- what is "executable" is of nobodies business but the interpreter loop
- we provide constants for the 7-bit `opCodeMask` and the derived `numOpCodes` for convenience to implement property maps like the static gas or the stack usage map

I think this further simplifies the definition. In particular, by dropping the distinction between valid and invalid op-codes, some functions also get rid of error cases. I prepared a PR targeting this branch show-casing how this is playing out. Let me know what you think.

Comment (LuisPH3) [reply go/interpreter/lfvm/opcode.go:376]: Unfortunately not all opcodes in rage 0-512 are valid.  there are gaps in the definition with values that are neither an op-code nor INVALID opcode.

**Comment by LuisPH3** [comment]
Note, I was not happy with last night performance runs, I repeated the test and validated the commit hashes. 
- Left column corresponds to the head of the master branch at the time of the experiment
- Right column is the head of the branch for this PR.

File: go/interpreter/lfvm/opcode.go [hunk 3]
```diff
@@ -10,387 +10,235 @@
...
+	// Invalid instruction
+	INVALID = OpCode(vm.INVALID)
+)
+
+// The following constants define the extended set of OpCodes for the long-form
+// EVM.
+// These opcodes are specific to the long-form EVM and are not part of the
+// original EVM.
+const (
+	FIRST_LFVM_EXTENDED_OPCODE = 0x100
```
[the remark this answers is not in the document]
Comment (LuisPH3) [reply go/interpreter/lfvm/opcode.go]: Function was removed.

File: go/interpreter/lfvm/opcode.go [hunk 4]
```diff
@@ -497,3 +352,25 @@ func (o OpCode) decompose() []OpCode {
...
+		return vm.IsValid(vm.OpCode(op))
+	}
+	return op >= FIRST_LFVM_EXTENDED_OPCODE && op < END_LFVM_EXTENDED_OPCODES
+}
+
+// isExecutable returns true if the OpCode is an executable OpCode.
+// An executable OpCode is a valid OpCode that executes an operation; all valid
+// OpCodes with the exception of DATA, and NOOP are executable.
+// - JUMPDEST is an executable OpCode.
+// - INVALID is not an executable OpCode, since it is neither Valid
```
[the remark this answers is not in the document]
Comment (LuisPH3) [reply go/interpreter/lfvm/opcode.go]: Function was removed.

File: go/interpreter/lfvm/opcode.go (hunk 2 above)
Comment (LuisPH3) [reply go/interpreter/lfvm/opcode.go]: Function was removed.

File: go/interpreter/lfvm/interpreter_test.go [hunk 5]
```diff
@@ -232,141 +220,76 @@ func TestInterpreter_step_DetectsUpperStackLimitViolation(t *testing.T) {
 	}
 }
 
-// TODO: this needs to be eliminated
```
Comment (LuisPH3) [inline go/interpreter/lfvm/interpreter_test.go:235]: NOTE: although out of scope, it was quite difficult to refactor this unit test. The code has been deleted here, but a migration branch exists already:
https://github.com/Fantom-foundation/Tosca/pull/753

**Comment by LuisPH3** [comment]
New benchmarks:
```
goos: linux
goarch: amd64
pkg: github.com/Fantom-foundation/Tosca/go/integration_test/interpreter
cpu: Intel(R) Xeon(R) Platinum 8481C CPU @ 2.70GHz
                         │    main     │             opcodes#0              │             opcodes#1              │           opcodes_16b_addr            │
                         │   sec/op    │   sec/op     vs base               │   sec/op     vs base               │    sec/op      vs base                │
Inc/1/lfvm-8               1.674µ ± 0%   1.719µ ± 0%  +2.72% (p=0.000 n=20)   1.719µ ± 0%  +2.72% (p=0.000 n=20)    1.729µ ±  4%   +3.29% (p=0.000 n=20)
Inc/10/lfvm-8              1.671µ ± 0%   1.718µ ± 0%  +2.81% (p=0.000 n=20)   1.718µ ± 0%  +2.81% (p=0.000 n=20)    1.727µ ±  1%   +3.35% (p=0.000 n=20)
Fib/1/lfvm-8               1.434µ ± 0%   1.491µ ± 0%  +4.01% (p=0.000 n=20)   1.491µ ± 0%  +4.01% (p=0.000 n=20)    1.493µ ±  0%   +4.12% (p=0.000 n=20)
Fib/5/lfvm-8               9.085µ ± 1%   9.212µ ± 1%  +1.39% (p=0.000 n=20)   9.212µ ± 1%  +1.39% (p=0.000 n=20)    9.190µ ± 25%   +1.15% (p=0.000 n=20)
Fib/10/lfvm-8              96.36µ ± 2%   95.56µ ± 0%       ~ (p=0.597 n=20)   95.56µ ± 0%       ~ (p=0.597 n=20)   122.57µ ± 22%  +27.20% (p=0.005 n=20)
Fib/15/lfvm-8              1.047m ± 2%   1.049m ± 0%       ~ (p=0.086 n=20)   1.049m ± 0%       ~ (p=0.086 n=20)    1.054m ± 28%   +0.65% (p=0.001 n=20)
Fib/20/lfvm-8              11.80m ± 2%   11.59m ± 0%       ~ (p=0.102 n=20)   11.59m ± 0%       ~ (p=0.102 n=20)    11.66m ± 29%        ~ (p=0.134 n=20)
Sha3/1/lfvm-8              635.9n ± 0%   657.0n ± 1%  +3.32% (p=0.000 n=20)   657.0n ± 1%  +3.32% (p=0.000 n=20)    637.8n ±  0%   +0.31% (p=0.001 n=20)
Sha3/10/lfvm-8             1.941µ ± 0%   1.964µ ± 0%  +1.18% (p=0.000 n=20)   1.964µ ± 0%  +1.18% (p=0.000 n=20)    1.958µ ±  3%   +0.88% (p=0.000 n=20)
Sha3/100/lfvm-8            14.63µ ± 2%   14.57µ ± 0%  -0.40% (p=0.036 n=20)   14.57µ ± 0%  -0.40% (p=0.036 n=20)    19.09µ ± 23%  +30.48% (p=0.002 n=20)
Sha3/1000/lfvm-8           141.8µ ± 3%   138.0µ ± 0%       ~ (p=0.211 n=20)   138.0µ ± 0%       ~ (p=0.211 n=20)    139.9µ ± 32%        ~ (p=0.221 n=20)
Arith/1/lfvm-8             1.820µ ± 0%   1.903µ ± 0%  +4.56% (p=0.000 n=20)   1.903µ ± 0%  +4.56% (p=0.000 n=20)    1.896µ ±  5%   +4.15% (p=0.000 n=20)
Arith/10/lfvm-8            4.814µ ± 1%   5.006µ ± 0%  +3.99% (p=0.000 n=20)   5.006µ ± 0%  +3.99% (p=0.000 n=20)    4.930µ ± 18%   +2.41% (p=0.000 n=20)
Arith/100/lfvm-8           34.19µ ± 2%   35.43µ ± 0%  +3.62% (p=0.000 n=20)   35.43µ ± 0%  +3.62% (p=0.000 n=20)    53.17µ ± 35%  +55.53% (p=0.000 n=20)
Arith/280/lfvm-8           95.37µ ± 2%   98.20µ ± 0%  +2.97% (p=0.000 n=20)   98.20µ ± 0%  +2.97% (p=0.000 n=20)   146.74µ ± 35%  +53.87% (p=0.000 n=20)
Memory/1/lfvm-8            2.931µ ± 0%   3.067µ ± 0%  +4.66% (p=0.000 n=20)   3.067µ ± 0%  +4.66% (p=0.000 n=20)    3.029µ ±  3%   +3.34% (p=0.000 n=20)
Memory/10/lfvm-8           7.143µ ± 0%   7.440µ ± 1%  +4.16% (p=0.000 n=20)   7.440µ ± 1%  +4.16% (p=0.000 n=20)    7.257µ ± 23%   +1.59% (p=0.000 n=20)
Memory/100/lfvm-8          48.64µ ± 0%   50.54µ ± 1%  +3.91% (p=0.000 n=20)   50.54µ ± 1%  +3.91% (p=0.000 n=20)    51.02µ ± 33%   +4.90% (p=0.001 n=20)
Memory/1000/lfvm-8         457.0µ ± 0%   470.0µ ± 0%  +2.85% (p=0.000 n=20)   470.0µ ± 0%  +2.85% (p=0.000 n=20)    669.1µ ± 31%  +46.43% (p=0.003 n=20)
Memory/10000/lfvm-8        4.534m ± 1%   4.662m ± 0%  +2.81% (p=0.000 n=20)   4.662m ± 0%  +2.81% (p=0.000 n=20)    4.565m ±  0%   +0.67% (p=0.002 n=20)
Analysis/jumpdest/lfvm-8   425.4n ± 0%   437.6n ± 0%  +2.87% (p=0.000 n=20)   437.6n ± 0%  +2.87% (p=0.000 n=20)    433.5n ±  0%   +1.92% (p=0.000 n=20)
Analysis/stop/lfvm-8       425.8n ± 0%   437.4n ± 0%  +2.72% (p=0.000 n=20)   437.4n ± 0%  +2.72% (p=0.000 n=20)    432.9n ±  1%   +1.66% (p=0.000 n=20)
Analysis/push1/lfvm-8      427.1n ± 0%   439.9n ± 0%  +3.01% (p=0.000 n=20)   439.9n ± 0%  +3.01% (p=0.000 n=20)    433.1n ±  0%   +1.40% (p=0.000 n=20)
Analysis/push32/lfvm-8     422.7n ± 0%   437.8n ± 0%  +3.60% (p=0.000 n=20)   437.8n ± 0%  +3.60% (p=0.000 n=20)    433.1n ±  0%   +2.47% (p=0.000 n=20)
geomean                    12.52µ        12.81µ       +2.30%                  12.81µ       +2.30%                   13.67µ         +9.20%
```

The rightmost column is a version where the stack limits lookup has been changed from `[]{int, int}` to `[]{int16, int16}`, the change was done with the intention of reducing `satisfiesStackRequirements` execution time, but the outcome is difficult to explain. This change was removed from this PR.
image

**Comment by LuisPH3** [comment]
Benchmarks after `opcodeProperty` refactor:

```
goos: linux
goarch: amd64
pkg: github.com/Fantom-foundation/Tosca/go/integration_test/interpreter
cpu: Intel(R) Xeon(R) Platinum 8481C CPU @ 2.70GHz
                         │   main_20   │         unified_opcodes_20         │
                         │   sec/op    │   sec/op     vs base               │
Inc/1/lfvm-8               1.672µ ± 0%   1.723µ ± 0%  +3.05% (p=0.000 n=20)
Inc/10/lfvm-8              1.670µ ± 0%   1.721µ ± 0%  +3.02% (p=0.000 n=20)
Fib/1/lfvm-8               1.434µ ± 0%   1.489µ ± 0%  +3.84% (p=0.000 n=20)
Fib/5/lfvm-8               9.033µ ± 0%   9.186µ ± 0%  +1.69% (p=0.000 n=20)
Fib/10/lfvm-8              94.89µ ± 0%   95.34µ ± 0%  +0.47% (p=0.000 n=20)
Fib/15/lfvm-8              1.047m ± 0%   1.051m ± 0%  +0.45% (p=0.000 n=20)
Fib/20/lfvm-8              11.58m ± 0%   11.67m ± 0%  +0.79% (p=0.000 n=20)
Sha3/1/lfvm-8              636.1n ± 0%   642.3n ± 0%  +0.97% (p=0.000 n=20)
Sha3/10/lfvm-8             1.937µ ± 0%   1.966µ ± 0%  +1.47% (p=0.000 n=20)
Sha3/100/lfvm-8            14.55µ ± 0%   14.66µ ± 0%  +0.76% (p=0.000 n=20)
Sha3/1000/lfvm-8           137.5µ ± 0%   139.6µ ± 0%  +1.56% (p=0.000 n=20)
Arith/1/lfvm-8             1.823µ ± 0%   1.895µ ± 0%  +3.95% (p=0.000 n=20)
Arith/10/lfvm-8            4.812µ ± 0%   4.927µ ± 0%  +2.39% (p=0.000 n=20)
Arith/100/lfvm-8           34.17µ ± 0%   34.27µ ± 0%  +0.31% (p=0.000 n=20)
Arith/280/lfvm-8           95.20µ ± 0%   95.46µ ± 0%  +0.27% (p=0.000 n=20)
Memory/1/lfvm-8            2.942µ ± 0%   3.008µ ± 0%  +2.24% (p=0.000 n=20)
Memory/10/lfvm-8           7.121µ ± 0%   7.270µ ± 1%  +2.10% (p=0.000 n=20)
Memory/100/lfvm-8          48.58µ ± 0%   49.16µ ± 2%  +1.18% (p=0.000 n=20)
Memory/1000/lfvm-8         456.7µ ± 0%   459.7µ ± 0%  +0.64% (p=0.000 n=20)
Memory/10000/lfvm-8        4.538m ± 0%   4.555m ± 0%  +0.38% (p=0.001 n=20)
Analysis/jumpdest/lfvm-8   426.9n ± 0%   429.8n ± 0%  +0.67% (p=0.000 n=20)
Analysis/stop/lfvm-8       426.4n ± 0%   430.6n ± 0%  +0.96% (p=0.000 n=20)
Analysis/push1/lfvm-8      427.7n ± 0%   431.9n ± 0%  +0.98% (p=0.000 n=20)
Analysis/push32/lfvm-8     427.0n ± 0%   434.2n ± 0%  +1.70% (p=0.000 n=20)
geomean                    12.48µ        12.67µ       +1.49%
```

File: go/interpreter/lfvm/opcode.go [hunk 6]
```diff
@@ -218,179 +246,14 @@ const (
...
-	PREVRANDAO:  "PREVRANDAO",
-	GAS:         "GAS",
-	GASLIMIT:    "GASLIMIT",
-	CHAINID:     "CHAINID",
-	SELFBALANCE: "SELFBALANCE",
-	BASEFEE:     "BASEFEE",
-	BLOBHASH:    "BLOBHASH",
-	BLOBBASEFEE: "BLOBBASEFEE",
-
+var to_string = map[OpCode]string{
```
Comment (HerbertJordan) [inline go/interpreter/lfvm/opcode.go:256]: nit: our code style convention would require this to be a `toString` or `opCodeStrings`; we could also switch to the generator based `stringer` solution in a follow-up PR.

File: go/interpreter/lfvm/converter_test.go [hunk 7]
```diff
@@ -222,10 +222,10 @@ func TestConvertWithObserver_MapsEvmToLfvmPositions(t *testing.T) {
 	}
 
 	for _, p := range pairs {
-		in := vm.OpCode(code[p.evm])
-		want := op_2_op[in]
-		if vm.PUSH1 <= in && in <= vm.PUSH32 {
-			want = PUSH1 + OpCode(in-vm.PUSH1)
+		toscaOpCode := vm.OpCode(code[p.evm])
+		want := OpCode(toscaOpCode)
+		if vm.PUSH1 <= toscaOpCode && toscaOpCode <= vm.PUSH32 {
```
Comment (HerbertJordan) [inline go/interpreter/lfvm/converter_test.go:227]: This special case handling for PUSH operations is no longer needed.

**Review by HerbertJordan — CHANGES_REQUESTED** [review]
I really like how this is shaping up. I understand that the instruction tests are quite messy, and basically, as we already discussed, we might eventually just do a complete do-over. So do not get entangled to long into those.

Besides them, I added a few requests for changes - all I think minor things. Please have a look.

File: go/interpreter/lfvm/opcode.go [hunk 8]
```diff
@@ -497,3 +367,27 @@ func (o OpCode) decompose() []OpCode {
...
+// newOpCodeProperty creates a new OpCode property map.
+func newOpCodeProperty[T any](init func(op OpCode) T) opCodeProperty[T] {
+	lookup := [numOpCodes]T{}
+	for i := 0; i < numOpCodes; i++ {
+		lookup[i] = init(OpCode(i))
+	}
+	return opCodeProperty[T]{lookup}
+}
+
+func (p opCodeProperty[T]) get(op OpCode) T {
```
[the remark this answers is not in the document]
Comment (LuisPH3) [reply go/interpreter/lfvm/opcode.go:391]: *ptr? . 
I haven't noticed any performance decay, benchmarks attached.

[the remark this answers is not in the document]
Comment (LuisPH3) [reply go/interpreter/lfvm/opcode.go]: well, you are right
```
 $ benchstat value reference 
goos: linux
goarch: amd64
pkg: github.com/Fantom-foundation/Tosca/go/integration_test/interpreter
cpu: AMD Ryzen 7 PRO 7840U w/ Radeon 780M Graphics  
                  │     value     │              reference              │
                  │    sec/op     │   sec/op     vs base                │
Arith/280/lfvm-16   1835.33µ ± 4%   76.59µ ± 1%  -95.83% (p=0.000 n=10)

                  │    value     │              reference              │
                  │     B/op     │     B/op      vs base               │
Arith/280/lfvm-16   1.081Ki ± 6%   1.018Ki ± 0%  -5.87% (p=0.000 n=10)

                  │   value    │           reference            │
                  │ allocs/op  │ allocs/op   vs base            │
Arith/280/lfvm-16   6.000 ± 0%   6.000 ± 0%  ~ (p=1.000 n=10) ¹
¹ all samples are equal
```

[the remark this answers is not in the document]
Comment (LuisPH3) [reply go/interpreter/lfvm/opcode.go]: thanks, that would make it.

**Comment by LuisPH3** [comment]
Updated benchmark after change to ```func (p *opCodePropertyMap[T]) get(op OpCode) T {```

```
goos: linux
goarch: amd64
pkg: github.com/Fantom-foundation/Tosca/go/integration_test/interpreter
cpu: Intel(R) Xeon(R) Platinum 8481C CPU @ 2.70GHz
                         │   main_20   │         unified_opcodes_20         │
                         │   sec/op    │   sec/op     vs base               │
Inc/1/lfvm-8               1.672µ ± 0%   1.674µ ± 0%       ~ (p=0.149 n=20)
Inc/10/lfvm-8              1.670µ ± 0%   1.672µ ± 0%       ~ (p=0.941 n=20)
Fib/1/lfvm-8               1.434µ ± 0%   1.461µ ± 0%  +1.85% (p=0.000 n=20)
Fib/5/lfvm-8               9.033µ ± 0%   9.023µ ± 0%  -0.12% (p=0.007 n=20)
Fib/10/lfvm-8              94.89µ ± 0%   94.00µ ± 0%  -0.94% (p=0.000 n=20)
Fib/15/lfvm-8              1.047m ± 0%   1.035m ± 0%  -1.09% (p=0.000 n=20)
Fib/20/lfvm-8              11.58m ± 0%   11.45m ± 0%  -1.13% (p=0.000 n=20)
Sha3/1/lfvm-8              636.1n ± 0%   633.6n ± 0%  -0.41% (p=0.000 n=20)
Sha3/10/lfvm-8             1.937µ ± 0%   1.957µ ± 0%  +1.03% (p=0.000 n=20)
Sha3/100/lfvm-8            14.55µ ± 0%   14.50µ ± 0%  -0.38% (p=0.000 n=20)
Sha3/1000/lfvm-8           137.5µ ± 0%   137.6µ ± 0%  +0.09% (p=0.011 n=20)
Arith/1/lfvm-8             1.823µ ± 0%   1.824µ ± 0%       ~ (p=0.773 n=20)
Arith/10/lfvm-8            4.812µ ± 0%   4.834µ ± 0%  +0.47% (p=0.000 n=20)
Arith/100/lfvm-8           34.17µ ± 0%   34.20µ ± 0%  +0.09% (p=0.038 n=20)
Arith/280/lfvm-8           95.20µ ± 0%   95.29µ ± 0%       ~ (p=0.071 n=20)
Memory/1/lfvm-8            2.942µ ± 0%   2.948µ ± 0%  +0.22% (p=0.043 n=20)
Memory/10/lfvm-8           7.121µ ± 0%   7.163µ ± 0%  +0.59% (p=0.000 n=20)
Memory/100/lfvm-8          48.58µ ± 0%   48.48µ ± 0%  -0.21% (p=0.002 n=20)
Memory/1000/lfvm-8         456.7µ ± 0%   455.4µ ± 0%  -0.30% (p=0.000 n=20)
Memory/10000/lfvm-8        4.538m ± 0%   4.511m ± 0%  -0.58% (p=0.000 n=20)
Analysis/jumpdest/lfvm-8   426.9n ± 0%   425.8n ± 0%       ~ (p=0.109 n=20)
Analysis/stop/lfvm-8       426.4n ± 0%   427.6n ± 0%       ~ (p=0.171 n=20)
Analysis/push1/lfvm-8      427.7n ± 0%   428.3n ± 0%       ~ (p=0.713 n=20)
Analysis/push32/lfvm-8     427.0n ± 0%   427.3n ± 0%       ~ (p=0.457 n=20)
geomean                    12.48µ        12.48µ       -0.01%
```

**Review by HerbertJordan — APPROVED** [review]

**Review by facuMH — DISMISSED** [review]

**Review by facuMH — APPROVED** [review]

**Pull request merged**